packages/mobile/src/services/offline-downloader/offline-sync.ts

```diff
@@ -99,21 +99,22 @@ const syncFavorites = async () => {
     .map(([id, track]) => track.track_id)
 
   const oldTrackIds = new Set([...queuedTracks, ...cachedFavoritedTrackIds])
-  const newTrackIds = new Set(favoritedTrackIds)
-  const addedTrackIds = [...newTrackIds].filter(
-    (trackId) => !oldTrackIds.has(trackId)
+  const newTrackIds = new Set(favoritedTrackIds.map(({ trackId }) => trackId))
+  const addedTrackIds = [...favoritedTrackIds].filter(
+    ({ trackId }) => !oldTrackIds.has(trackId)
   )
   const removedTrackIds = [...oldTrackIds].filter(
     (trackId) => !newTrackIds.has(trackId)
   )
 
   const tracksForDownload: TrackForDownload[] = addedTrackIds.map(
     (addedTrack) => ({
-      trackId: addedTrack,
+      trackId: addedTrack.trackId,
```

Comment:
huh. Does `fetchAllFavoritedTrackIds` not return `number[]`? Why do we need `.trackId` here?

Reply:
should rename some vars. this now as id and created at